h2o-core/src/main/java/water/H2O.java

```diff
@@ -1965,6 +1992,49 @@ public void run() {
     }
   }
 
+  /**
+   * Register {@link NodeDisconnectHook}
+   * @param hook to register
+   */
+  public static void registerNodeDisconnectedHook(NodeDisconnectHook hook){
+    nodeDisconnectHooks.add(hook);
+  }
+
+  /**
+   *   Thread used to run disconnect hooks on nodes who disconnects from the cloud
+   */
+  public static class DisconnectedNodesWatchdog extends Thread {
+    final private int sleepMillis = 6000; // 6 seconds
```

Comment:
just had a look on example you provided - https://github.com/h2oai/h2o-3/blob/master/h2o-grpc/src/main/java/ai/h2o/api/GrpcExtension.java.

In that case I would go with regular h2o argument which can be specified as part of the extension